src/preferences/dialog/dlgpreferences.cpp

```diff
@@ -371,12 +373,29 @@ void DlgPreferences::onShow() {
         screenSpace = QSize(800, 600);
     }
     else {
-        screenSpace = pScreen->size();
+        screenSpace = pScreen->availableSize();
     }
-    newX = std::max(0, std::min(newX, screenSpace.width() - m_geometry[2].toInt()));
-    newY = std::max(0, std::min(newY, screenSpace.height() - m_geometry[3].toInt()));
+
+    // Make sure the entire window is visible on screen and is not occluded by taskbar
+    // Note: Window geometry excludes window decoration
+    int windowDecorationWidth = frameGeometry().width() - geometry().width();
+    int windowDecorationHeight = frameGeometry().height() - geometry().height();
+    if (windowDecorationWidth <= 0) {
+        windowDecorationWidth = 2;
+    }
+    if (windowDecorationHeight <= 0) {
+        windowDecorationHeight = 30;
+    }
+    int availableWidth = screenSpace.width() - windowDecorationWidth;
+    int availableHeight = screenSpace.height() - windowDecorationHeight;
+    newWidth = std::min(newWidth, availableWidth);
+    newHeight = std::min(newHeight, availableHeight);
+    newX = std::max(0, std::min(newX, availableWidth - newWidth));
+    newY = std::max(0, std::min(newY, availableHeight - newHeight));
```

Comment:
https://en.cppreference.com/w/cpp/algorithm/clamp
```suggestion
    newX = std::clamp(availableWidth - newWidth, 0, newX);
    newY = std::clamp(availableHeight - newHeight, 0, newY);
```